xgi/classes/reportviews.py

```diff
@@ -237,7 +237,22 @@ def filterby_attr(self, attr, val, mode="eq"):
         <https://github.com/ComplexGroupInteractions/xgi/blob/main/tutorials/Tutorial%206%20-%20Statistics.ipynb>`_.
 
         """
-        bunch = [idx for idx in self if self._id_attr[idx][attr] == val]
+        if mode == "eq":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] == val]
+        elif mode == "neq":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] != val]
+        elif mode == "lt":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] < val]
+        elif mode == "gt":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] > val]
+        elif mode == "leq":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] <= val]
+        elif mode == "geq":
+            bunch = [idx for idx in self if self._id_attr[idx][attr] >= val]
+        elif mode == "between":
+            bunch = [idx for idx in self if val[0] <= self._id_attr[idx][attr] <= val[1]]
+        else:
+            raise ValueError(f"Unrecognized mode {mode}")
```

Comment:
Would specify which modes are known:

raise ValueError(f"Unrecognized mode {mode}. mode must be one of ["eq", "neq", "lt", "gt", "leq", "geq", "between"].")

Reply:
Good call. Fixed here and also in `filterby`